Sources/MapboxMaps/Viewport/States/FollowingViewportState.swift

```diff
@@ -0,0 +1,138 @@
+public struct FollowingViewportStateOptions: Hashable {
+    public var zoom: CGFloat
+    public var pitch: CGFloat
+
+    public init(zoom: CGFloat = 15,
+                pitch: CGFloat = 40) {
+        self.zoom = zoom
+        self.pitch = pitch
+    }
+}
+
+public final class FollowingViewportState {
+
+    // MARK: - Public Config
+
+    public var options: FollowingViewportStateOptions {
+        didSet {
+            processUpdatedCamera()
+        }
+    }
+
+    // MARK: - Injected Dependencies
+
+    private let cameraAnimationsManager: CameraAnimationsManagerProtocol
+
+    // MARK: - Private Dependencies
+
+    // avoids needing to expose a NSObject subclass in the public API.
+    // this can be improved in the next major version by eliminating
+    // the requirement that `LocationConsumer`s inherit from NSObject.
+    private let delegatingLocationConsumer: DelegatingLocationConsumer
+
+    // MARK: - Private State
+
+    private var latestLocation: Location? {
+        didSet {
+            processUpdatedCamera()
+        }
+    }
+
+    private var observers = [CameraObserver]()
+
+    private var isUpdatingCamera = false
+
+    private var cameraAnimationCancelable: Cancelable?
+
+    // MARK: - Initialization
+
+    internal init(options: FollowingViewportStateOptions,
+                  locationProducer: LocationProducerProtocol,
+                  cameraAnimationsManager: CameraAnimationsManagerProtocol) {
+        self.options = options
+        self.cameraAnimationsManager = cameraAnimationsManager
+        self.delegatingLocationConsumer = DelegatingLocationConsumer(locationProducer: locationProducer)
+        self.delegatingLocationConsumer.delegate = self
+    }
+
+    // MARK: - Private Utilities
+
+    private func cameraOptions(for location: Location) -> CameraOptions {
+        return CameraOptions(
+            center: location.location.coordinate,
+            zoom: options.zoom,
+            bearing: location.location.course,
+            pitch: options.pitch)
+    }
+
+    private func animate(to cameraOptions: CameraOptions) {
+        cameraAnimationCancelable?.cancel()
+        cameraAnimationCancelable = cameraAnimationsManager.ease(
```

Comment:
Navigation SDK for iOS relies on `BasicCameraAnimator`, when it comes to viewport transitions. Do you plan to rely on easing functions only, or this part is still under development? There might be a need to use different timing parameters, delays etc.

Reply:
This was under development. I've now ported the transition logic from Android. See DefaultViewportTransition and related classes.